mzmine-community/src/main/java/io/github/mzmine/modules/batchmode/autosave/AutoSaveBatchTask.java

```
@@ -0,0 +1,133 @@
+  protected @NotNull List<RawDataFile> getProcessedDataFiles() {
+    return List.of();
+  }
+
+  @Override
+  protected void process() {
+    if (!TaskService.getController().isTaskInstanceRunningOrQueued(BatchTask.class)) {
+      // don't error since we don't want to stop processing
+      DialogLoggerUtil.showDialog(AlertType.WARNING, "Cannot save batch",
+          "Cannot save current batch since no batch is running. Please consider saving manually.");
+      return;
+    }
+
+    final BatchTask batchTask = TaskService.getController().getReadOnlyTasks().stream()
+        .filter(task -> task.getActualTask() instanceof BatchTask)
+        .map(task -> (BatchTask) task.getActualTask()).findFirst().orElse(null);
+    assert batchTask != null;
+
+    final BatchQueue queueCopy = batchTask.getQueueCopy();
+
+    final String mzbatch = ExtensionFilters.getExtensionName(ExtensionFilters.MZ_BATCH);
+
+    File savePath = parameters.getOptionalValue(AutoSaveBatchParameters.savePath).orElse(null);
+    if (savePath == null || savePath.toString().isBlank()) {
+//      Set.of(SiriusExportModule.class, GnpsFbmnExportAndSubmitModule.class,
+//          CSVExportModularModule.class, LegacyCSVExportModule.class,
+//          ProjectMetadataExportModule.class);
+      final File commonFromExport = ParameterUtils.extractCommonExportPath(queueCopy);
+      final File commonImport = ParameterUtils.extractCommonRawFileImportFilePath(queueCopy);
+      if (commonFromExport != null) {
+        savePath = new File(commonFromExport, "batch.%s".formatted(mzbatch));
+      } else if (commonImport != null) {
+        savePath = new File(commonImport, "batch.%s".formatted(mzbatch));
+      }
+    }
+
+    if (savePath == null || savePath.toString().isBlank()) {
+      DialogLoggerUtil.showDialog(AlertType.WARNING, "Cannot save batch",
+          "The %s module cannot automatically determine a path to export the batch file to. Please save the batch manually.",
+          false);
```

maybe better to just error() on this task and let the batch handle the dialog?

---

The idea is to not let the batch fail

---

Ah ok. Makes sense. 
Then maybe add a comment not that some smart ass will change that in the future

---

Are you talking about yourself? :D